tests/Feature/Proxy/CarrierProxyTest.php

```diff
@@ -31,7 +31,7 @@ public function setUp()
 
         $this->client = $this->getClientMock();
         $this->authenticator = $this->getAuthenticatorMock();
-        $this->api = (new MyParcelComApi('https://api'))
+        $this->api = (new MyParcelComApi('https://api', $this->client))
             ->setCache(new NullCache())
             ->setHttpClient($this->client)
```

Comment:
You're setting the same client twice, once in the constructor and once with the setter.